dispatch/dispatch.go

```diff
@@ -404,55 +393,52 @@ func (ag *aggrGroup) stop() {
 
 // insert inserts the alert into the aggregation group.
 func (ag *aggrGroup) insert(alert *types.Alert) {
-	ag.mtx.Lock()
-	defer ag.mtx.Unlock()
-
-	ag.alerts[alert.Fingerprint()] = alert
+	ag.alerts.Set(alert)
 
 	// Immediately trigger a flush if the wait duration for this
 	// alert is already over.
+	ag.mtx.Lock()
+	defer ag.mtx.Unlock()
 	if !ag.hasFlushed && alert.StartsAt.Add(ag.opts.GroupWait).Before(time.Now()) {
 		ag.next.Reset(0)
 	}
 }
 
 func (ag *aggrGroup) empty() bool {
-	ag.mtx.RLock()
-	defer ag.mtx.RUnlock()
-
-	return len(ag.alerts) == 0
+	// TODO: Does this need to be drained?
```

Comment:
We're returning `len(ag.alerts.List())`, which is a buffered channel -- I'm not sure if we need to drain all the alerts in the channel, or if we can just leave it and the garbage collector realizes that there are no references to the channel, so it can all be cleaned up.